resources/js/Components/LoadingOverlay.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+    <div class="loading-indicator" v-if="shown">
+        <div class="progressbar" role="progressbar" />
+        <div class="overlay" />
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+    name: 'LoadingOverlay',
+    props: {
+        delay: {
+            type: Number,
+            default: 250,
+        },
+        visible: {
+            type: Boolean,
+            default: false,
+        }
+    },
+    data() {
+        return {
+            shown: this.visible,
+            document: {
+                overflow: 'auto'
+            }
+        };
+    },
+    methods: {
+        show(): void {
+            if (this.shown) {
+                return;
+            }
+
+            const bodyStyles = window.getComputedStyle(document.body);
+            this.shown = true;
+
+            this.document.overflow = bodyStyles.overflow;
```

Comment:
It works great. Thanks. I am having trouble understanding why this part was implemented like this

Reply:
I want to avoid adding an extra css class to an html element that is not is the scope of this template, so We change the overflow programmatically. To do so, we cache this value so that we don't accidentally override any styles that were applied to the document in html, css or other scripts. This is so that the component itself is not too closely coupled to the current implementation (which implicitly assumes that we actually can override the body's overflow with just a class).

Reply:
Yes, it's a good solution to go back to whatever styles where applied before.